internal/cli/atlas/clusters/advancedsettings/update.go

```diff
@@ -0,0 +1,194 @@
+		return err
+	}
+}
+
+func (opts *UpdateOpts) Run() error {
+	r, err := opts.store.UpdateAtlasClusterConfigurationOptions(opts.ConfigProjectID(), opts.name, opts.newProcessArgs())
+	if err != nil {
+		return commonerrors.Check(err)
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newProcessArgs() *atlas.ProcessArgs {
+	args := &atlas.ProcessArgs{}
+
+	if opts.defaultReadConcern != defaultReadConcern {
+		args.DefaultReadConcern = opts.defaultReadConcern
+	}
+
+	if opts.defaultWriteConcern != defaultWriteConcern {
+		args.DefaultWriteConcern = opts.defaultWriteConcern
+	}
+
+	if opts.minimumEnabledTLSProtocol != "" {
+		args.MinimumEnabledTLSProtocol = opts.minimumEnabledTLSProtocol
+	}
+
+	if opts.sampleSizeBIConnector != defaultSampleSizeBIConnector {
+		args.SampleSizeBIConnector = pointy.Int64(opts.sampleSizeBIConnector)
+	}
+
+	if opts.sampleRefreshIntervalBIConnector != 0 {
+		args.SampleRefreshIntervalBIConnector = pointy.Int64(opts.sampleRefreshIntervalBIConnector)
+	}
+
+	if opts.disableTableScan {
+		args.NoTableScan = pointy.Bool(opts.disableTableScan)
+	}
+
+	if opts.enableTableScan {
+		args.NoTableScan = pointy.Bool(!opts.enableTableScan)
+	}
+
+	if opts.disableJavascript {
+		args.JavascriptEnabled = pointy.Bool(false)
+	}
+
+	if opts.enableJavascript {
+		args.JavascriptEnabled = pointy.Bool(true)
+	}
+
+	if opts.disableFailIndexKeyTooLong {
+		args.FailIndexKeyTooLong = pointy.Bool(false)
+	}
+
+	if opts.enableFailIndexKeyTooLong {
+		args.FailIndexKeyTooLong = pointy.Bool(true)
+	}
+
+	if opts.oplogSizeMB != 0 {
+		args.OplogSizeMB = pointy.Int64(opts.oplogSizeMB)
+	}
+
+	if opts.oplogMinRetentionHours != 0 {
+		args.OplogMinRetentionHours = pointy.Float64(opts.oplogMinRetentionHours)
+	}
+
+	return args
+}
+
+// atlas cluster(s) advancedSettings update <clusterName> --projectId projectId [--readConcern readConcern]
+// [--writeConcern writeConcern] [--disableFailIndexKeyTooLong true/fale] [--enableFailIndexKeyTooLong true/fale] [--disableJavascript true/false] [--enableJavascript true/false]
+// [--minimumEnabledTLSProtocol minimumEnabledTLSProtocol] [--disableTableScan true/false] [--enableTableScan true/false] [--oplogMinRetentionHours oplogMinRetentionHours]
+// [--oplogSizeMB oplogSizeMB] [--sampleRefreshIntervalBIConnector sampleRefreshIntervalBIConnector] [--sampleSizeBIConnector sampleSizeBIConnector].
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{}
+	cmd := &cobra.Command{
+		Use:   "update <clusterName>",
+		Short: "Update advanced configuration settings for one cluster.",
+		Example: fmt.Sprintf(`  Update oplog minimum size for a cluster:
+  $ %[1]s cluster advancedSettings update <clusterName> --projectId <projectId> --oplogSizeMB 1000
+
+  Update minimum TLS protocol for a cluster:
+  $ %[1]s cluster advancedSettings update <clusterName> --projectId <projectId> --minimumEnabledTLSProtocol "TLS1_0"`,
```

Comment:
suggestion: I think TLS 1.0 is deprecated and known to be insecure. consider using a different version in the example 
```suggestion
  $ %[1]s cluster advancedSettings update <clusterName> --projectId <projectId> --minimumEnabledTLSProtocol "TLS1_2"`,
```

Reply:
updated